snuba/datasets/errors_replacer.py

```diff
@@ -761,9 +761,25 @@ def process_unmerge_hierarchical(
         ),
     }
 
-    # Sentry is expected to send an `exclude_groups` message after unsplit is
-    # done, and we can live with data inconsistencies while this is ongoing.
-    query_time_flags = (None, message["project_id"])
+    if message.get("skip_needs_final", False):
+        # In some cases Sentry can ensure that no events remain in
+        # `previous_group_id`. In that case FINAL is not applied to future
+        # queries, Sentry is instead expected to send an `exclude_groups`
+        # message after unsplit is done, and we can live with data
+        # inconsistencies while this is ongoing.
+        #
+        # Concretely:
+        #
+        # * Every time the grouping level is increased and one group is split
+        #   up into multiple, skip_needs_final can be used.
+        #
+        # * Every time this process needs to be reverted (merging parts of many
+        #   groups into one group), it cannot be used. Since this is "undo"
+        #   like regular unmerge is "undo" of merge, it is expected to happen
+        #   way less often.
```

Comment:
I am not sure I follow this paragraph. `skip_needs_final`, when used, is the problem. Though we say it cannot be used on the operation that happens less often. Do you mean "it does not need to be used" ?

Reply:
I elaborated in the code comment such that we have some permanent documentation + example and rephrased this part.